File: 2-add-two-numbers/2-add-two-numbers.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode addTwoNumbers(ListNode l1, ListNode l2) {
        if (l1 == null && l2 == null) 
            return null;
		else if (l1 == null) 
            return l2; 
		else if (l2 == null) 
            return l1; 

		int a = l1.val + l2.val;
		ListNode p = new ListNode(a % 10);
		p.next = addTwoNumbers(l1.next,l2.next);
        
		if (a >= 10) 
            p.next = addTwoNumbers(p.next, new ListNode(1));
        return p;
    }
}

File: 2-add-two-numbers/2-add-two-numbers.py
# Definition for singly-linked list.
class ListNode(object):
    def __init__(self, val=0, next=None):
        self.val = val
        self.next = next
        
class Solution:
    def addTwoNumbers(self, l1, l2):
        if (l1 == None and l2 == None):
            return None
        elif (l1 == None):
            return l2
        elif (l2 == None):
            return l1
        
        a = l1.val + l2.val
        p = ListNode(a % 10)
        p.next = self.addTwoNumbers(l1.next, l2.next)
        
        if (a >= 10):
            p.next = self.addTwoNumbers(p.next, ListNode(1))
        return p